app/routes/goal_routes.py

```diff
@@ -1 +1,74 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, make_response, abort, request, Response
+from app.models.goal import Goal
+from app.models.task import Task
+from ..db import db
+from .route_utilities import create_model, validate_model
+
+bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@bp.post("")
+def create_goal():
+    request_body = request.get_json()
+    return create_model(Goal, request_body)
+
+@bp.get("")
+def get_all_goals():
+    query = db.select(Goal).order_by(Goal.id)
+    goals = db.session.scalars(query)
+    return [goal.to_dict()["goal"] for goal in goals]
+
+@bp.get("/<goal_id>")
+def get_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    return goal.to_dict(), 200
+
+@bp.put("/<goal_id>")
+def update_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    return goal.to_dict()
+
+@bp.delete("/<goal_id>")
+def delete_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response({"details": f"Goal {goal_id} \"{goal.title}\" successfully deleted"})
+
+@bp.get("/<goal_id>/tasks")
+def get_all_tasks_associated_with_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    tasks_response = [task.to_dict()["task"] for task in goal.tasks]
+    
+    response = {
+        "id": goal.id,
+        "title": goal.title,
+        "tasks": tasks_response
+    }
+
+    return response
+
+@bp.post("/<goal_id>/tasks")
+def create_task_specific_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+    task_ids = request_body.get("task_ids", [])
+
+    for task in task_ids:
+        Task.query.get(task).goal_id = goal.id
```

Comment:
I missed this earlier in my review @PriyankaKaramchandani 

The syntax `<model>.query.get` has been deprecated since SQLAlchemy 2.0 and **should not be used for new development**. Latest syntax would look like `db.session.query`.

Better yet, you would use the `validate_model` function here:
```py
for task_id in task_ids:
        task = validate_model(Task, task_id)
        goal.tasks.append(task)